### PR TITLE
Switch from useIsSimpleSeller hook to getSiteOption

### PR DESCRIPTION
**Pull request by markbiek**
#### Changes proposed in this Pull Request

The hook isn't necessary. All we need to do is use `getSiteOption` to check if the site intent is `sell` on the WooCommerce landing page (the only place the hook was being used).

#### Testing instructions

Visit a non-seller website admin and go to `/woocommerce-installation`. You should see the following content:

image

Then visit the `/woocommerce-installation` page for a website that used the "Start simple" seller signup flow. You should see the following content instead:

image

Related to #62031

### CODE REVIEW

**Review by nelsonec87 — APPROVED** [review]
Worked as described!

**Comment by markbiek** [comment]
Thanks @nelsonec87 !

@andres-blanco Would you mind giving this a quick 👀 when you have a second? Just since you are familiar with the other, related PR?

File: client/my-sites/woocommerce/landing-page.tsx [hunk 1]
```diff
@@ -38,7 +39,7 @@ const LandingPage: React.FunctionComponent< Props > = ( { siteId } ) => {
 	const { __ } = useI18n();
 	const navigationItems = [ { label: 'WooCommerce' } ];
 	const ctaRef = useRef( null );
-	const simpleSeller = useIsSimpleSeller();
```
Comment (andres-blanco) [inline client/my-sites/woocommerce/landing-page.tsx:41]: I've seen this used differently.
https://github.com/Automattic/wp-calypso/blob/f2e6f88ae65c8571bbac21b284c29b48ec987598/client/my-sites/checkout/composite-checkout/components/prepurchase-notices/jetpack-plugin-required-version-notice.jsx#L47-L49

I think the current usage is fine but I find the one I linked a little more concise. Anyway I'll leave the choice up to you 😄 .

**Review by andres-blanco — APPROVED** [review]
This is working fine! I left a note but nothing blocking

File: client/my-sites/woocommerce/landing-page.tsx (hunk 1 above)
Comment (markbiek) [reply client/my-sites/woocommerce/landing-page.tsx:41]: Oh, I like that. Much tidier! I'll get that updated shortly.

Thanks for the tip!

**Pull request merged**